homeassistant/components/homepilot/cover.py

```diff
@@ -0,0 +1,91 @@
+"""Support for HomePilot Covers."""
+
+import asyncio
+
+from pyhomepilot.devices import Blind
+from pyhomepilot.utils import reverse_percentage
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GARAGE,
+    DEVICE_CLASS_SHADE,
+    DEVICE_CLASS_WINDOW,
+    CoverEntity,
+)
+
+from .const import DOMAIN, STOP_UPDATE_DELAY
+from .entity import HomePilotEntity
+
+# supported device classes based on selected icon set
+DEVICE_CLASSES = {
+    "iconset6": DEVICE_CLASS_BLIND,
+    "iconset7": DEVICE_CLASS_SHADE,
+    "iconset8": DEVICE_CLASS_SHADE,
+    "iconset9": DEVICE_CLASS_DOOR,
+    "iconset12": DEVICE_CLASS_DOOR,
+    "iconset14": DEVICE_CLASS_DOOR,
+    "iconset15": DEVICE_CLASS_SHADE,
+    "iconset20": DEVICE_CLASS_BLIND,
+    "iconset24": DEVICE_CLASS_WINDOW,
+    "iconset30": DEVICE_CLASS_GARAGE,
+    "iconset31": DEVICE_CLASS_GARAGE,
+    "iconset34": DEVICE_CLASS_WINDOW,
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up HomePilot cover platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for uid, device in instance["coordinator"].data.items():
+        if isinstance(device, Blind):
+            entities.append(HomePilotCoverEntity(instance, uid))
+
+    async_add_entities(entities)
+
+
+class HomePilotCoverEntity(HomePilotEntity, CoverEntity):
+    """Class representing HomePilot cover."""
+
+    @property
+    def device_class(self):
+        """Return the class of the device."""
+        if not self._device.iconSet["k"] in DEVICE_CLASSES:
+            return None
+        return DEVICE_CLASSES[self._device.iconSet["k"]]
```

Comment:
```suggestion
        return DEVICE_CLASSES.get(self._device.iconSet["k"])
```
since `get()` defaults to `None` :wink: